bin/varnishd/mgt/mgt_jail_linux.c

```diff
@@ -49,10 +49,60 @@
 #include "common/heritage.h"
 
 static int
-vjl_init(char **args)
+vjl_set_thp(const char *arg)
 {
+	if (!strcmp(arg, "enable")) {
+		if (prctl(PR_SET_THP_DISABLE, 0, 0, 0, 0) != 0) {
+			MGT_Complain(C_INFO,
+			    "Could not enable Transparent Hugepages: %s (%d)",
+			    VAS_errtxt(errno), errno);
+		}
+	} else if (!strcmp(arg, "disable") ||
+		    !strcmp(arg, "default")) {
+		if (prctl(PR_SET_THP_DISABLE, 1, 0, 0, 0) != 0) {
+			MGT_Complain(C_INFO,
+			    "Could not disable Transparent Hugepages: %s (%d)",
+			    VAS_errtxt(errno), errno);
+		}
+	} else {
+		MGT_Complain(C_ERR, "Invalid value for \"transparent_hugepages\""
+		    " jail option");
+		return (1);
+	}
+	return (0);
+}
 
-	return jail_tech_unix.init(args);
+static int
+vjl_init(char **args)
+{
+	char **uargs = NULL;
+	size_t i = 0;
+	int ret = 0;
+
+	if (args != NULL) {
+		for (; args[i] != NULL; i++);
+		uargs = malloc(sizeof(char*) * (i + 1));
```

Comment:
I'd be open for another solution in order to separate Linux jail's parameters from the Unix one, this one does not look pretty imho but I havent found another (simple and elegant) way to do it

Reply:
You can pass them one by one with a 2-length array living on the stack to avoid keeping track of all the non-linux arguments.

Reply:
Actually no, you can't, the UNIX jail init must happen in a single call.